types/vcpkg-definition.json

```diff
@@ -0,0 +1,50 @@
+{
+  "$schema": "https://packageurl.org/schemas/purl-type-definition.schema-1.0.json",
+  "$id": "https://packageurl.org/types/vcpkg-definition.json",
+  "type": "vcpkg",
+  "type_name": "Vcpkg C/C++ packages",
+  "description": "Vcpkg C/C++ packages",
+  "repository": {
+    "use_repository": true,
+    "default_repository_url": "https://github.com/microsoft/vcpkg/",
+    "note": "The absolute URL of the vcpkg registry where the package is available (optional). If omitted, ``https://github.com/microsoft/vcpkg`` as default registry is assumed. For filesystem registries or [overlay ports](https://learn.microsoft.com/vcpkg/concepts/overlay-ports), the URI will have a `file` URI scheme."
+  },
+  "namespace_definition": {
+    "requirement": "prohibited",
+    "note": "Currently reserved for future use and should be empty."
+  },
+  "name_definition": {
+    "requirement": "required",
+    "native_name": "package-name",
+    "note": "The vcpkg package name."
+  },
+  "version_definition": {
+    "requirement": "required",
+    "native_name": "package-version",
+    "note": "The vcpg package version."
```

Comment:
There seems to be cases where the port-version is appended to the package-version as explained in https://github.com/azure-sdk/vcpkg
> Minimum required version. Port-version is identified with a #N suffix, for example, 1.2.3#7 names port-version 7.

Is this only the case sometimes, or often enough?
Could this mean that the port-version would be always part of the package version, and not a separate qualifier?